src/main/java/calculator/NonNegativeTokens.java

```diff
@@ -0,0 +1,31 @@
+package calculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class NonNegativeTokens {
+  private static final String EMPTY_STRING = "";
+
+  private List<NonNegativeInt> integers;
+
+  public NonNegativeTokens(String expression, String delimiter) {
+    if (expression == null || expression.equals(EMPTY_STRING)) {
+      integers = new ArrayList<>(Arrays.asList(new NonNegativeInt(0)));
+      return;
+    }
+
+    integers = Arrays.stream(expression.split(delimiter))
+        .map(Integer::parseInt)
+        .map(NonNegativeInt::new)
+        .collect(Collectors.toList());
+  }
```

Comment:
```suggestion
  public NonNegativeTokens(String expression, String delimiter) {
    this(extract(expression, delimiter));
  }

  private static List<NonNegativeInt> extract(String expression, String delimiter) {
    if (isNullOrEmpty(expression)) {
      return getDefaultValue();
    }

    return split(expression, delimiter);
  }
```

이번 클린코드 과정에서는 메서드를 극단적으로 나눠보는 연습을 해보는 것도 좋다고 생각합니다.

Reply:
메서드를 극단적으로 나눠보는 연습을 하면 어떤 점이 좋은가요?(이 점을 알면 더 도움이 될 것 같습니다)